src/plot.jl

```diff
@@ -35,6 +35,19 @@ function plot_states(ρ;order=:numeric,spin_comp=ones(Int(log2(size(ρ)[1]))),nu
         error("order must be either :numeric or :hamming or :prob")
     end
 
+    kept_states = 0:2^n-1
+    if energy_levels != 0
+        if ising_model == nothing
+            error("must provide ising_model if energy levels are specified")
+        else
+            energies = _QA.compute_ising_energy_levels(ising_model)
+            kept_states = foldl(union, [energies[i].states for i = 1:energy_levels])
+        end
+    end
+
+    filter_function = (x) -> _QA.spin2int(x.spin_vec) in kept_states
+    states = filter(filter_function, states)
+
```

Comment:
Given this similar processing occurs in at least two functions, I am wondering if you should break it out as an internal function like `kept_states = _get_states(ising_model, energy_levels)` which each function will call?

Reply:
@ccoffrin I think this was a good idea.  I went ahead and added it, and it was a lot more finicky than I expected with all the edge cases.  I ended up just testing every possible input case to make sure it behaved as expected and there are no gaps in  the control flow/logic